stl/inc/iterator

```diff
@@ -807,36 +814,38 @@ public:
     constexpr common_iterator(_Se _Right) noexcept(is_nothrow_move_constructible_v<_Se>) // strengthened
         : _Val{in_place_type<_Se>, _STD move(_Right)} {}
 
+    constexpr explicit common_iterator(_Common_iterator_construct_tag _Tag) noexcept : _Val{_Tag} {}
+
     // clang-format off
-    template <class _Iter2, class _Se2>
-        requires convertible_to<const _Iter2&, _Iter> && convertible_to<const _Se2&, _Se>
-    constexpr common_iterator(const common_iterator<_Iter2, _Se2>& _Right) noexcept(
-        is_nothrow_constructible_v<_Iter, const _Iter2&>&& is_nothrow_constructible_v<_Se, const _Se2&>) // strengthened
-        : _Val{_Right._Val} {}
-
-    template <class _Iter2, class _Se2>
-        requires convertible_to<const _Iter2&, _Iter> && convertible_to<const _Se2&, _Se>
-              && assignable_from<_Iter&, const _Iter2&> && assignable_from<_Se&, const _Se2&>
-    common_iterator& operator=(const common_iterator<_Iter2, _Se2>& _Right) noexcept(
-        is_nothrow_assignable_v<_Iter, const _Iter2&>&& is_nothrow_assignable_v<_Se, const _Se2&>&&
-            is_nothrow_constructible_v<_Iter, const _Iter2&>&&
-                is_nothrow_constructible_v<_Se, const _Se2&>) /* strengthened */ {
+    template <class _OIter, class _OSe>
+        requires convertible_to<const _OIter&, _Iter> && convertible_to<const _OSe&, _Se>
+    constexpr common_iterator(const common_iterator<_OIter, _OSe>& _Right) noexcept(
+        is_nothrow_constructible_v<_Iter, const _OIter&>&& is_nothrow_constructible_v<_Se, const _OSe&>) // strengthened
+        : _Val{_Vty{_Right._Val}} {}

```

Comment:
This did not compile for me as the `_Variantish` constructor is explicit

```cpp
template <class _OIter, class _OSe>
        requires _Not_same_as<_Variantish<_OIter, _OSe>, _Variantish>
    explicit _Variantish(const _Variantish<_OIter, _OSe>& _That)
```

Reply:
`_Val{_Right._Val}` performs *direct-initialization*, so it can call `explicit` constructors - it must have failed for some other reason. In any case, I removed the explicit conversion and the tests pass - but not with `abort()` in the body of this constructor, so it's definitely being used.